### PR TITLE
[Main Course] - Notes of rules of ML2.0

### PR DESCRIPTION
**Pull request by jhqian0909**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Post for notes of rules of ML2.0

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

# How Has This Been Tested?

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/Main_Course/Data_Analytics/2021-07-24-notes-of-rules-of-ML2.0.md [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+Rule #29: The best way to make sure that you train like you serve is to save the set of features used at serving time, and then pipe those features to a log to use them at training time.
+```
+
+最好的方法是将真实线上的数据log下来作为训练数据。这可以大大减少两者之间的差异性。
+
+```
+Rule #30: Importance weight sampled data, don’t arbitrarily drop it!
+```
+
+不要随意丢弃采样样本。同时，对采样概率为30%的样本，训练时要给10/3的权重。这种校准对基于预测值的模型很重要，对基于排序的模型影响不大。
```
Comment (Realvincentyuan) [inline docs/Main_Course/Data_Analytics/2021-07-24-notes-of-rules-of-ML2.0.md:121]: 看了一下原文，意思是降采样的数据，的权重也应该相应的降低，是应该这么理解吗？
Screen Shot 2021-07-31 at 4 13 44 PM

**Review by Realvincentyuan — APPROVED** [review]

File: docs/Main_Course/Data_Analytics/2021-07-24-notes-of-rules-of-ML2.0.md (hunk 1 above)
Comment (jhqian0909) [reply docs/Main_Course/Data_Analytics/2021-07-24-notes-of-rules-of-ML2.0.md:121]: 感觉是采样率是30%的样本要给一个更大的权重(10/3)

File: docs/Main_Course/Data_Analytics/2021-07-24-notes-of-rules-of-ML2.0.md [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+Rule #33: If you produce a model based on the data until January 5th, test the model on the data from January 6th and after.
+```
+
+测试数据要在训练数据之后。
+
+```
+Rule #34: In binary classification for filtering (such as spam detection or determining interesting e-mails), make small short-term sacrifices in performance for very clean data.
+```
+
+如果你已经有选择的给用户展示邮件，那么你的训练数据将是有偏的。比较好的做法是用一个没有任何干扰的control group作为训练集。不需要太大，1%，2%就好。
```
Comment (jz263) [inline docs/Main_Course/Data_Analytics/2021-07-24-notes-of-rules-of-ML2.0.md:145]: 这里看第一遍有点没看懂，感觉可以围绕如何sacrifice解释一下：优化上线后的二元分类模型时，可以牺牲小部分(1%)流量，让其不经过模型直接展现给用户，这样既可以保证现有模型的上线效果不受严重影响，又可以获得足够数量的无偏样本对模型进行进一步优化。

Comment (jhqian0909) [reply docs/Main_Course/Data_Analytics/2021-07-24-notes-of-rules-of-ML2.0.md]: 已修改。这部分control group一般是用来训练模型的:)

**Pull request merged**